traits/traits_listener.py

```diff
@@ -990,20 +981,50 @@ def parse(self):
                     handler=self.handler,
                     wrapped_handler_ref=self.wrapped_handler_ref,
                     dispatch=self.dispatch,
+                    priority=self.priority,
+                    # Bug-for-bug compatibility with old behaviour: don't
+                    # propagate the 'deferred' or 'handler_type' values for the
+                    # child item.
+                    deferred=False,
+                    type=ANY_LISTENER,
                 ),
                 handler=self.handler,
                 wrapped_handler_ref=self.wrapped_handler_ref,
                 dispatch=self.dispatch,
+                priority=self.priority,
+                deferred=deferred,
+                type=handler_type,
             )
 
-        return self.parse_group(EOS)
+        return self.parse_group(
+            terminator=EOS,
+            deferred=deferred,
+            handler_type=handler_type,
+        )
 
-    def parse_group(self, terminator="]"):
```

Comment:
Not entirely sure why the default for `terminator` changed here either.

Looking at changes later in the PR (where `]` is explicitly passed as `terminator`), it looks like `]` should have never been the default. Is that correct?

Reply:
While refactoring, I found it helpful to remove the default altogether from `parse_group` and make it a required argument, so that the code explicitly specified the default in each call. The actual behaviour shouldn't have changed here - where we were relying on the default previously, we're now explicitly passing `"]"`; where we were using something different, we're passing it as before.

If you prefer, I can revert this change and reintroduce the default and make `terminator` optional again. I find it easier to understand the code with `terminator` a required argument.